File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/TmdbImage.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Represents image metadata returned by TMDB.
 *
 * @param aspectRatio image aspect ratio
 * @param height      image height in pixels
 * @param iso6391     optional ISO 639-1 language code
 * @param filePath    TMDB image file path
 * @param voteAverage average image vote
 * @param voteCount   image vote count
 * @param width       image width in pixels
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TmdbImage(
        @JsonProperty("aspect_ratio")
        double aspectRatio,

        int height,

        @JsonProperty("iso_639_1")
        String iso6391,

        @JsonProperty("file_path")
        String filePath,

        @JsonProperty("vote_average")
        double voteAverage,

        @JsonProperty("vote_count")
        int voteCount,

        int width
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/TmdbImage.kt
package dev.reuss.tmdb.common.image

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Represents image metadata returned by TMDB.
 *
 * @property aspectRatio image aspect ratio
 * @property height      image height in pixels
 * @property iso6391 ISO 639-1 language code associated with the value.
 * @property filePath TMDB image path for this image.
 * @property voteAverage Average user rating for the image reported by TMDB.
 * @property voteCount Number of user ratings for the image reported by TMDB.
 * @property width       image width in pixels
 */
@JvmRecord
data class TmdbImage(
    @all:JsonProperty("aspect_ratio")
    val aspectRatio: Double,
    @all:JsonProperty("height")
    val height: Int,
    @all:JsonProperty("iso_639_1")
    val iso6391: String?,
    @all:JsonProperty("file_path")
    val filePath: String?,
    @all:JsonProperty("vote_average")
    val voteAverage: Double,
    @all:JsonProperty("vote_count")
    val voteCount: Int,
    @all:JsonProperty("width")
    val width: Int,
) : TmdbModel
